src/controller/CHIPDeviceController.h

```diff
@@ -349,7 +356,7 @@ class DLL_EXPORT DeviceController : public AbstractDnssdDiscoveryController
      *          It can be used for fine-grained dependency injection of a controller's
      *          NOC and operational keypair.
      */
-    CHIP_ERROR InitControllerNOCChain(const ControllerInitParams & params);
+    CHIP_ERROR InitControllerNOCChain(const ControllerInitParams & params, bool UpdateNOCOnly = false);
```

Comment:
It seems to me that the public API should be the existing InitControllerNOCChain and a new UpdateControllerNOCChain, and the thing with the extra arg should be an internal-only method.

That said, a few other questions:

1. The NOC update bits should presumably close existing sessions if the node ID is changing right?  They don't seem to do that.
2. Is the right argument to "update NOC" really ControllerInitParams?   That struct has a bunch of stuff that has nothing to do with the NOC chain.  Perhaps we should have a new struct that has the NOC bits and ControllerInitParams should inherit from it or something?

Reply:
Have added a new UpdateControllerNOCChain function.